thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StructContext.java

```diff
@@ -132,35 +138,85 @@ protected TField getTFieldByName(String name) throws TException {
      * TProtocol.writeStructBegin(), rather than relying on the stack trace.
      */
     private static Class<?> getCurrentThriftMessageClass() {
-        final StackTraceElement[] frames =
-                Thread.currentThread().getStackTrace();
-
-        for (StackTraceElement f : frames) {
-            final String className = f.getClassName();
-
+        if (SystemInfo.javaVersion() >= 9) {
```

Comment:
Instead of doing this on every `getCurrentThriftMessageClass()`, we could detect the Java version only once at the class initializer (`static` block), e.g.

```java
final class StructContext ... {
    private static final Supplier<Class<?>> currentThriftMessageClass;
    static {
        Supplier<Class<?>> supplier = null;
        if (SystemInfo.javaVersion() >= 9) {
            try {
                supplier = new StackWalkingCurrentThriftMessageClassFinder();
            } catch (Throwable t) {
                log.warn(...);
            }
        }

        currentThriftMessageClass = supplier != null ? supplier : new DefaultCurrentThriftMessageClassFunder();
        }
    }
    ...
}
```